openpgp/packet/config.go

```diff
@@ -33,16 +35,24 @@ type Config struct {
 	DefaultCompressionAlgo CompressionAlgo
 	// CompressionConfig configures the compression settings.
 	CompressionConfig *CompressionConfig
-	// S2KCount is only used for symmetric encryption. It
-	// determines the strength of the passphrase stretching when
+	// S2K (String to Key) config, used for key derivation in the context of secret key encryption
+	// and password-encrypted data.
+	// If nil, the default configuration is used
+	S2KConfig *s2k.Config
+	// Iteration count for Iterated S2K (String to Key). 
+	// Only relevant if sk2.Mode is set to s2k.IteratedSaltedS2K.
```

Comment:
This sort of implies that if you pass `S2KConfig: &s2k.Config{ Mode: IteratedSaltedS2K }` that it's still used, but it's not, so I'd say:
```suggestion
	// Only used if sk2.Mode is nil.
```